frontend/src/pages/queryBuilder/textEditor/textEditorRow/NodeSelector.jsx

```diff
@@ -115,6 +123,15 @@ export default function NodeSelector({
     }
   }, [open, searchTerm]);
 
+  /**
+   * Cancel any api calls on unmount
```

Comment:
I'm not clear on when this is supposed to happen. Does the component get unmounted whenever the user changes what is typed in the search box?

Reply:
No, this happens when the user navigates to a different page.

Reply:
Ah, I was confused, the issue is solved at line 103.